### DISCUSSION TITLE
`belongsToThrough` can be implemented by inverting the foreign keys on a `hasOneThrough` relationship

### DISCUSSION THREAD

**Discussion by gbalduzzi** [laravel/framework#52386] [Ideas]
I recently needed to create a through relationship for this simple structure:

```
Document --- belongsTo ---> Project --- belongsTo ---> User
```

I wanted to define a relationship between Document and User.

I expected a `belongsToThrough` relationship, but I see there is not one available.

It was discussed in #6161 but it didn't end up being implemented.

The best relationship available for this use case is `HasOneThrough` but **it is defined backwards**.
 It could be used to link User to a Document, assuming we know that both are 1 - 1 relationships (not the case here).

This problem was already noted in #42114, that also includes a viable solution to reach the desired result: just manually define the key constraints, reversing them from their defaults.  In my case, this is:

```
return $this->hasOneThrough(
    User::class,
    Project::class,
    'id', // local key on the projects table
    'id', // local key on the users table
    'project_id', // foreign key on the documents table
    'user_id', // foreign key on the projects table
```

----

Given the realization that a proper belongsToThrough relationship can be implemented without much effort reusing the HasOneThrough existing relationship, is there interest in implementing it?

Am I missing something, and this is not a fully viable solution?

I can submit a PR but given how the previous issues were closed I'm not sure there in interest or desire in it.

**Comment by ghost** [comment]
belongs to and has one can be used for the same thing. So hasOneThrough is usable for your case even if it is not named according to what you want.

Can you please give arguments against the above? Maybe I am wrong.

**Reply by gbalduzzi** [reply]
Yes, belongsToThrough can be achieved manually through hasOneThrough. But I don't see why this is a valid reason not to implement the proper belongsToThrough. 

Quite the contrary:

- Effort is minimal, given the fact that the logic is already there in the `hasOneThrough` existing implementation
- It is not immediately clear that it can be accomplished already. I suspect many developers tried using the `hasOneThrough` and gave up after a while
- It is the only kind of relationship where the default naming convention is not consistent

**Reply by ghost** [reply]
Thank you.

**Reply by ghost** [reply]
HasMany/OneDeep or HasMany/OneThrough 2 or 3 link tables should also be implemented but they are not(except for external packages)
